lib/cosmos/tools/launcher/launcher_tool.rb

```diff
@@ -53,10 +53,27 @@ def parameters_dialog
       @variable_parameters.each do |parameter_name, parameter_value|
         hlayout = Qt::HBoxLayout.new
         hlayout.addWidget(Qt::Label.new(parameter_name))
-        line_edit = Qt::LineEdit.new()
-        line_edit.setText(parameter_value)
-        hlayout.addWidget(line_edit)
-        widgets << line_edit
+        parameter_options = parameter_value.split('|')
+        if parameter_options.length()>1
```

Comment:
Don't use parens () with methods with no parameters unless you are calling a method implicitely on self.  

Is: 
if parameter_options.length()>1

Should be:
if parameter_options.length > 1

There are a few other places to such as calls to new() should just be .new.   (I know the old code was bad too... my ruby style has adapted over the years)